tests/filecheck/transforms/riscv_lower_parallel_mov.mlir

```diff
@@ -92,16 +92,74 @@ builtin.module {
+  %5, %6, %7, %8, %9, %10, %11, %12, %13, %14 = riscv.parallel_mov %2, %0, %1, %0, %1, %2, %4, %3, %3, %3 {free_registers = [!riscv.reg<s11>]} : (!riscv.reg<s3>, !riscv.reg<s1>, !riscv.reg<s2>, !riscv.reg<s1>, !riscv.reg<s2>, !riscv.reg<s3>, !riscv.reg<s8>, !riscv.reg<s7>, !riscv.reg<s7>, !riscv.reg<s7>) -> (!riscv.reg<s1>, !riscv.reg<s2>, !riscv.reg<s3>, !riscv.reg<s4>, !riscv.reg<s5>, !riscv.reg<s6>, !riscv.reg<s7>, !riscv.reg<s8>, !riscv.reg<s9>, !riscv.reg<s10>)
+  "test.op"(%5, %6, %7, %8, %9, %10, %11, %12, %13, %14) : (!riscv.reg<s1>, !riscv.reg<s2>, !riscv.reg<s3>, !riscv.reg<s4>, !riscv.reg<s5>, !riscv.reg<s6>, !riscv.reg<s7>, !riscv.reg<s8>, !riscv.reg<s9>, !riscv.reg<s10>) -> ()
+}
+// CHECK:       builtin.module {
+// CHECK-NEXT:    %0, %1, %2, %3, %4 = "test.op"() : () -> (!riscv.reg<s1>, !riscv.reg<s2>, !riscv.reg<s3>, !riscv.reg<s7>, !riscv.reg<s8>)
+// CHECK-NEXT:    %5 = riscv.mv %0 : (!riscv.reg<s1>) -> !riscv.reg<s4>
+// CHECK-NEXT:    %6 = riscv.mv %1 : (!riscv.reg<s2>) -> !riscv.reg<s5>
+// CHECK-NEXT:    %7 = riscv.mv %2 : (!riscv.reg<s3>) -> !riscv.reg<s6>
+// CHECK-NEXT:    %8 = riscv.mv %3 : (!riscv.reg<s7>) -> !riscv.reg<s9>
+// CHECK-NEXT:    %9 = riscv.mv %3 : (!riscv.reg<s7>) -> !riscv.reg<s10>
+// CHECK-NEXT:    %10 = riscv.mv %2 : (!riscv.reg<s3>) -> !riscv.reg<s11>
+// CHECK-NEXT:    %11 = riscv.mv %1 : (!riscv.reg<s2>) -> !riscv.reg<s3>
+// CHECK-NEXT:    %12 = riscv.mv %0 : (!riscv.reg<s1>) -> !riscv.reg<s2>
+// CHECK-NEXT:    %13 = riscv.mv %10 : (!riscv.reg<s11>) -> !riscv.reg<s1>
+// CHECK-NEXT:    %14 = riscv.mv %4 : (!riscv.reg<s8>) -> !riscv.reg<s11>
+// CHECK-NEXT:    %15 = riscv.mv %3 : (!riscv.reg<s7>) -> !riscv.reg<s8>
+// CHECK-NEXT:    %16 = riscv.mv %14 : (!riscv.reg<s11>) -> !riscv.reg<s7>
+// CHECK-NEXT:    "test.op"(%13, %12, %11, %5, %6, %7, %16, %15, %8, %9) : (!riscv.reg<s1>, !riscv.reg<s2>, !riscv.reg<s3>, !riscv.reg<s4>, !riscv.reg<s5>, !riscv.reg<s6>, !riscv.reg<s7>, !riscv.reg<s8>, !riscv.reg<s9>, !riscv.reg<s10>) -> ()
+// CHECK-NEXT:  }
+
+// -----
+
+// Test no free registers
+//    s1
+//   /  ^
+//   |  |
+//   v  /
+//    s2
 builtin.module {
   %0, %1 = "test.op"() : () -> (!riscv.reg<s1>, !riscv.reg<s2>)
   %2, %3 = riscv.parallel_mov %0, %1 : (!riscv.reg<s1>, !riscv.reg<s2>) -> (!riscv.reg<s2>, !riscv.reg<s1>)
   "test.op"(%2, %3) : (!riscv.reg<s2>, !riscv.reg<s1>) -> ()
 }
 
 // CHECK:         %2, %3 = "riscv.parallel_mov"(%0, %1) : (!riscv.reg<s1>, !riscv.reg<s2>) -> (!riscv.reg<s2>, !riscv.reg<s1>)
-// CHECK-NEXT:    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^--------------------------------
-// CHECK-NEXT:    | Error while applying pattern: Not implemented: cyclic moves
-// CHECK-NEXT:    -------------------------------------------------------------
+// CHECK-NEXT:    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^-----------------------------------------------
+// CHECK-NEXT:    | Error while applying pattern: Cyclic move detected with no free registers.
```

Comment:
We should handle moves when there are no registers free, would be great to add it in the next PR, should probably change this:
```suggestion
// CHECK-NEXT:    | Error while applying pattern: Not implemented: cyclic moves without free int registers.
```